### PR TITLE
refactor: Simplify api.py (move endpoints to their own routers)

### PR DESCRIPTION
**Pull request by ArturLange**
### What

Move endpoints to different directory to clean up `api.py` 

See https://fastapi.tiangolo.com/tutorial/bigger-applications/

### CODE REVIEW

File: app/views/auth.py
Comment (raphodn) [inline app/views/auth.py:1]: Thanks @ArturLange for starting this refactoring work. I have a question on the approach : why the `views` folder naming ? did you use a specific tutorial as reference ?

Comment (raphodn) [reply app/views/auth.py:1]: I did a bit of research and I found 

1)  the fasapi doc : https://fastapi.tiangolo.com/tutorial/bigger-applications/
which favors naming them `routes` + extracting the route prefix

2) a reddit question : that favors the Django approach of having 1 folder per app

Comment (raphodn) [reply app/views/auth.py:1]: being a Django dev (and slightly regretting not having chosen Django for this backend ^^), I kinda favor that approach, though we're probably not there yet in terms of complexity. so maybe go with the "official" routing approach for now ?

Comment (ArturLange) [reply app/views/auth.py:1]: I agree with the `routes` point, I can change that. For the "one folder per app" thing - AFAIK in Django "app" isn't what we usually mean - one web application can have many "apps" and I don't think that applies in FastAPI.
I'm against having `api.py` file with thousands of lines of code - it's hard to look at and it makes it harder to change something without creating conflicts with other people.

Comment (raphodn) [reply app/views/auth.py:1]: Thanks for the answer :)

Actually the Django approach would indeed mean multiple folders *inside* the app folder.

Same as you, the api.py has got too long and it's a good time to refactor 👍

Comment (raphodn) [reply app/views/auth.py:1]: @ArturLange done ? there's a CI that is failing :)

File: app/routes/auth.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        if session:
+            return crud.update_session_last_used_field(db, session=session)
+    raise HTTPException(
+        status_code=status.HTTP_401_UNAUTHORIZED,
+        detail="Invalid authentication credentials",
+        headers={"WWW-Authenticate": "Bearer"},
+    )
+
+
+@auth_router.post("", tags=["Auth"])
```
Comment (raphodn) [inline app/routes/auth.py:49]:
```suggestion
@auth_router.post("")
```
no need to repeat the `tags` info now that it is defined at the top :)

File: app/views/auth.py
Comment (raphodn) [reply app/views/auth.py:1]: Ok I did some additional cleanup :+1:

Comment (ArturLange) [reply app/views/auth.py:1]: Thank you!
Sorry for the delay on my side

**Pull request merged**